In [1]:
import pandas as pd
import sqlite3

# Read the CSV file into a DataFrame
df = pd.read_csv('country.csv')

# Create a connection to the SQLite database
conn = sqlite3.connect('database.db')

# Use the DataFrame.to_sql() method to create a new table in the database and populate it with the data from the DataFrame
df.to_sql('country', conn, if_exists='replace', index=False)



237673

In [2]:

df = pd.read_csv('match.csv')
df.to_sql('match', conn, if_exists='replace', index=False)


237673

In [3]:
df = pd.read_csv('medal_type.csv')
df.to_sql('medal_type', conn, if_exists='replace', index=False)

237673

In [4]:
df = pd.read_csv('player.csv')
df.to_sql('player', conn, if_exists='replace', index=False)

237673

In [5]:
%load_ext sql
%sql sqlite:///database.db

# This is a historical dataset on the modern Olympic Games, including all the Games from Athens 1896 to Tokyo 2020.


# This dataset contains 237673 rows and we splitted it to 4 tables.
## Now we want to run some simple SQL queries.

### 1. SQL query that selects all the distinct values of the "Sport" column 

In [6]:
%%sql
select distinct(sport)
from match
order by sport

*  sqlite:///database.db
Done.


[('3x3 Basketball',),
 ('Aeronautics',),
 ('Alpinism',),
 ('Archery',),
 ('Art Competitions',),
 ('Artistic Gymnastics',),
 ('Artistic Swimming',),
 ('Athletics',),
 ('Badminton',),
 ('Baseball',),
 ('Baseball/Softball',),
 ('Basketball',),
 ('Basque Pelota',),
 ('Beach Volleyball',),
 ('Boxing',),
 ('Canoe Slalom',),
 ('Canoe Sprint',),
 ('Canoeing',),
 ('Cricket',),
 ('Croquet',),
 ('Cycling',),
 ('Cycling BMX Freestyle',),
 ('Cycling BMX Racing',),
 ('Cycling Mountain Bike',),
 ('Cycling Road',),
 ('Cycling Track',),
 ('Diving',),
 ('Equestrian',),
 ('Equestrianism',),
 ('Fencing',),
 ('Figure Skating',),
 ('Football',),
 ('Golf',),
 ('Gymnastics',),
 ('Handball',),
 ('Hockey',),
 ('Ice Hockey',),
 ('Jeu De Paume',),
 ('Judo',),
 ('Karate',),
 ('Lacrosse',),
 ('Marathon Swimming',),
 ('Modern Pentathlon',),
 ('Motorboating',),
 ('Polo',),
 ('Racquets',),
 ('Rhythmic Gymnastics',),
 ('Roque',),
 ('Rowing',),
 ('Rugby',),
 ('Rugby Sevens',),
 ('Sailing',),
 ('Shooting',),
 ('Skateboarding',),
 ('Softball',),
 ('Sport Climbing',),
 ('Surfing',),
 ('Swimming',),
 ('Synchronized Swimming',),
 ('Table Tennis',),
 ('Taekwondo',),
 ('Tennis',),
 ('Trampoline Gymnastics',),
 ('Trampolining',),
 ('Triathlon',),
 ('Tug-Of-War',),
 ('Volleyball',),
 ('Water Polo',),
 ('Weightlifting',),
 ('Wrestling',)]


### 2. SQL query that selects maximum and minimum ages of female players

In [6]:
%%sql
select max(age),min(age)
from Player
where Sex='F' 
order by Age desc
limit 5;

*  sqlite:///database.db
Done.


[(74.0, 11.0)]

### 3. SQL query that selects total medals by year.

In [7]:
%%sql
select count(*) ,medal,year
from Medal_type
where medal is not null
group by medal,year
order by year desc
limit 20;

*  sqlite:///database.db
Done.


[(867, 'Bronze', 2020),
 (800, 'Gold', 2020),
 (782, 'Silver', 2020),
 (703, 'Bronze', 2016),
 (665, 'Gold', 2016),
 (655, 'Silver', 2016),
 (679, 'Bronze', 2012),
 (632, 'Gold', 2012),
 (630, 'Silver', 2012),
 (710, 'Bronze', 2008),
 (671, 'Gold', 2008),
 (667, 'Silver', 2008),
 (677, 'Bronze', 2004),
 (664, 'Gold', 2004),
 (660, 'Silver', 2004),
 (680, 'Bronze', 2000),
 (663, 'Gold', 2000),
 (661, 'Silver', 2000),
 (629, 'Bronze', 1996),
 (608, 'Gold', 1996)]

### 4. SQL query that selects count of events by year

In [8]:
%%sql
select year,count(event)
from Match,Medal_type 
where match.player_id=Medal_type.player_id
group by year
order by year desc
limit 20;

*  sqlite:///database.db
Done.


[(2020, 28004),
 (2016, 14591),
 (2012, 13722),
 (2008, 14394),
 (2004, 14215),
 (2000, 14636),
 (1996, 14585),
 (1992, 13714),
 (1988, 12799),
 (1984, 10070),
 (1980, 7665),
 (1976, 9137),
 (1972, 10961),
 (1968, 9139),
 (1964, 8137),
 (1960, 8609),
 (1956, 5372),
 (1952, 8711),
 (1948, 6816),
 (1936, 6803)]

### 5. SQL query that selects the winers that recieved gold medals in 1996. return the name of participants,the countries and the events.

In [9]:
%%sql
select Name,team,Event
from country,Medal_type,Match,Player
where country.team_id =Medal_type.player_id
And Medal_type.player_id=match.player_id
And match.player_id=player.player_id
And medal='Gold'
and year=1996;

*  sqlite:///database.db
Done.


[('AGUIAR Mayra', 'Brazil', 'Mixed Team'),
 ('Jos Mara Abarca Plotas', 'Brazil', 'Mixed Team'),
 ('AGUIAR Mayra', 'Brazil', "Water Polo Men's Water Polo"),
 ('Jos Mara Abarca Plotas', 'Brazil', "Water Polo Men's Water Polo"),
 ('AGUIAR Mayra', 'Spain', 'Mixed Team'),
 ('Jos Mara Abarca Plotas', 'Spain', 'Mixed Team'),
 ('AGUIAR Mayra', 'Spain', "Water Polo Men's Water Polo"),
 ('Jos Mara Abarca Plotas', 'Spain', "Water Polo Men's Water Polo"),
 ('AIT DAOUD Kamel', 'Algeria', "Lightweight Men's Double Sculls Team"),
 ('Agostino Abbagnale', 'Algeria', "Lightweight Men's Double Sculls Team"),
 ('AIT DAOUD Kamel', 'Algeria', "Rowing Men's Double Sculls"),
 ('Agostino Abbagnale', 'Algeria', "Rowing Men's Double Sculls"),
 ('AIT DAOUD Kamel', 'Italy', "Lightweight Men's Double Sculls Team"),
 ('Agostino Abbagnale', 'Italy', "Lightweight Men's Double Sculls Team"),
 ('AIT DAOUD Kamel', 'Italy', "Rowing Men's Double Sculls"),
 ('Agostino Abbagnale', 'Italy', "Rowing Men's Double Sculls"),
 ('BRNOVIC Tatjana', 'France-1', "Canoeing Men's Canadian Doubles, Slalom"),
 ('Franck Adisson', 'France-1', "Canoeing Men's Canadian Doubles, Slalom"),
 ('BRNOVIC Tatjana', 'France-1', 'Women Team'),
 ('Franck Adisson', 'France-1', 'Women Team'),
 ('BRNOVIC Tatjana', 'Montenegro', "Canoeing Men's Canadian Doubles, Slalom"),
 ('Franck Adisson', 'Montenegro', "Canoeing Men's Canadian Doubles, Slalom"),
 ('BRNOVIC Tatjana', 'Montenegro', 'Women Team'),
 ('Franck Adisson', 'Montenegro', 'Women Team'),
 ('BROMELL Trayvon', 'United States', "Athletics Men's 400 metres Hurdles"),
 ('Derrick Ralph Adkins', 'United States', "Athletics Men's 400 metres Hurdles"),
 ('BROMELL Trayvon', 'United States', "Men's 100m"),
 ('Derrick Ralph Adkins', 'United States', "Men's 100m"),
 ('BROMELL Trayvon', 'United States', "Athletics Men's 400 metres Hurdles"),
 ('Derrick Ralph Adkins', 'United States', "Athletics Men's 400 metres Hurdles"),
 ('BROMELL Trayvon', 'United States', "Men's 100m"),
 ('Derrick Ralph Adkins', 'United States', "Men's 100m"),
 ('Andre Kirk Agassi', 'Italy', "Tennis Men's Singles"),
 ('CARINI Angela', 'Italy', "Tennis Men's Singles"),
 ('Andre Kirk Agassi', 'Italy', "Women's Welter (64-69kg)"),
 ('CARINI Angela', 'Italy', "Women's Welter (64-69kg)"),
 ('Andre Kirk Agassi', 'United States', "Tennis Men's Singles"),
 ('CARINI Angela', 'United States', "Tennis Men's Singles"),
 ('Andre Kirk Agassi', 'United States', "Women's Welter (64-69kg)"),
 ('CARINI Angela', 'United States', "Women's Welter (64-69kg)"),
 ('DAL MOLIN Paolo', 'Cuba', "Baseball Men's Baseball"),
 ('Omar Ajete Iglesias', 'Cuba', "Baseball Men's Baseball"),
 ('DAL MOLIN Paolo', 'Cuba', "Men's 110m Hurdles"),
 ('Omar Ajete Iglesias', 'Cuba', "Men's 110m Hurdles"),
 ('DAL MOLIN Paolo', 'Italy', "Baseball Men's Baseball"),
 ('Omar Ajete Iglesias', 'Italy', "Baseball Men's Baseball"),
 ('DAL MOLIN Paolo', 'Italy', "Men's 110m Hurdles"),
 ('Omar Ajete Iglesias', 'Italy', "Men's 110m Hurdles"),
 ('Chioma Ajunwa (-Opara)', 'Brazil', "Athletics Women's Long Jump"),
 ("D'ALMEIDA Marcus", 'Brazil', "Athletics Women's Long Jump"),
 ('Chioma Ajunwa (-Opara)', 'Brazil', "Men's Individual"),
 ("D'ALMEIDA Marcus", 'Brazil', "Men's Individual"),
 ('Chioma Ajunwa (-Opara)', 'Nigeria', "Athletics Women's Long Jump"),
 ("D'ALMEIDA Marcus", 'Nigeria', "Athletics Women's Long Jump"),
 ('Chioma Ajunwa (-Opara)', 'Nigeria', "Men's Individual"),
 ("D'ALMEIDA Marcus", 'Nigeria', "Men's Individual"),
 ('DAWSON Jacob', 'Great Britain', "Football Women's Football"),
 ('Michelle Anne Akers', 'Great Britain', "Football Women's Football"),
 ('DAWSON Jacob', 'Great Britain', "Men's Eight Team"),
 ('Michelle Anne Akers', 'Great Britain', "Men's Eight Team"),
 ('DAWSON Jacob', 'United States', "Football Women's Football"),
 ('Michelle Anne Akers', 'United States', "Football Women's Football"),
 ('DAWSON Jacob', 'United States', "Men's Eight Team"),
 ('Michelle Anne Akers', 'United States', "Men's Eight Team"),
 ('IVOVI

### 6. SQL query that selects the first and last years of summer olympics.

In [12]:
%%sql
select min(year) as first_year, max(year) as last_year
from Medal_type;

*  sqlite:///database.db
Done.


[(1896, 2020)]

### 7. SQL query that selects the average age of female and male athletes who won a medal?

In [10]:
%%sql
select sex, AVG(age) AS avg_age
from Player,Medal_type
where Player.Player_id=Medal_type.Player_id
and medal IS NOT NULL
GROUP BY sex

*  sqlite:///database.db
Done.


[('F', 24.670602125147578), ('M', 26.541530293112228)]

### 8. SQL query that selects the event in which women have participated most?

In [11]:
%%sql
select sex,event,count(sex) as max,Year
from Player,Match,Medal_type 
where Player.player_id=Match.player_id
and Player.player_id=Medal_type.player_id
and Sex='F'
group by Event
Order by max desc
limit 1;

*  sqlite:///database.db
Done.


[('F', 'Women Team', 2925, 2012)]

### 9. SQL query that selects top 5 sports in which female athletes are more interested ?

In [12]:
%%sql
select sport, COUNT(medal) AS medal_count,sex
from Match,Medal_type,player
where Match.player_id=Medal_type.player_id
and Match.player_id=player.player_id
and sex = 'F'
Group by sport
ORDER BY medal_count DESC
LIMIT 5;

*  sqlite:///database.db
Done.


[('Athletics', 2037, 'F'),
 ('Swimming', 1860, 'F'),
 ('Rowing', 969, 'F'),
 ('Gymnastics', 898, 'F'),
 ('Hockey', 664, 'F')]

### 10. SQL query that selects the team which has won the most gold medal.return the count of medals and the year

In [18]:
%%sql
select team,COUNT(Medal) AS mdal_won, year
from country,Medal_type
where country.team_id=Medal_type.player_id
and medal='Gold'
GROUP BY team
ORDER BY medal_won DESC
LIMIT 1

*  sqlite:///database.db
Done.


[('United States', 2503, 1900)]